pkg/serverless/trace/propagation/carriers.go

```diff
@@ -203,6 +216,29 @@ func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
 	return carrier, nil
 }
 
+// handleEventBridgeThroughSNS builds payload carrier for an EventBridge
+// event wrapped in an SNS message.
+func handleEventBridgeThroughSNS(eventBridgeEvent map[string]interface{}) (tracer.TextMapReader, error) {
+	detail, ok := eventBridgeEvent[detailHeader].(map[string]interface{})
+	if !ok {
+		return nil, errorUnsupportedPayloadType
+	}
+
+	traceContextAny, ok := detail[datadogTraceHeader].(map[string]interface{})
+	if !ok {
+		return nil, errorNoDDContextFound
+	}
+
+	// Some values are integers, not strings, so we have to convert before using tracer.TextMapCarrier()
+	traceContext := make(map[string]string)
+	for key, val := range traceContextAny {
+		traceContext[key] = fmt.Sprintf("%v", val)
```

Comment:
Also, below in the `eventBridgeCarrier` we do not look for int fields like you are doing here. Should we be?